<?xml version="1.0"?>
<launch>

  <!-- Global Planner -->
  <arg name="base_global_planner" default="glwoa_rrtstar_planner/GlWoaRrtStarPlanner"/>
  <!-- <arg name="base_global_planner" default="global_planner/GlobalPlanner"/> -->
  <!-- <arg name="base_global_planner" default="navfn/NavfnROS"/> -->
  
  <!-- Local Planner -->
  <arg name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>
  <!-- <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/> -->
  <!-- <arg name="base_local_planner" default="teb_local_planner/TebLocalPlannerROS"/> -->
 
  
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    
    <rosparam file="$(find glwoa_rrtstar_planner)/config/common_costmap.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find glwoa_rrtstar_planner)/config/common_costmap.yaml" command="load" ns="local_costmap" />
    
    <rosparam file="$(find glwoa_rrtstar_planner)/config/local_costmap.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find glwoa_rrtstar_planner)/config/global_costmap.yaml" command="load" ns="global_costmap" /> 

    <!-- Local planner parameters -->
    <!-- <rosparam file="$(find glwoa_rrtstar_planner)/config/local_planner_teb_default.yaml" command="load"/> -->
    <rosparam file="$(find glwoa_rrtstar_planner)/config/local_planner_trajectoryPlanner.yaml" command="load"/>
    <!-- <rosparam file="$(find glwoa_rrtstar_planner)/config/local_planner_dwa.yaml" command="load"/> -->
   
    <!-- Global planner parameters -->
    <rosparam file="$(find glwoa_rrtstar_planner)/config/glwoa_rrtstar_planner.yaml" command="load"/>
    <!-- <rosparam file="$(find glwoa_rrtstar_planner)/config/global_planner.yaml" command="load"/> -->
    <!-- <rosparam file="$(find glwoa_rrtstar_planner)/config/global_planner_navfn.yaml" command="load"/> -->
  
  </node>

</launch>